packages/muelu/src/Transfers/Petrov-Galerkin-SA/MueLu_PgPFactory_def.hpp

```diff
@@ -214,6 +214,14 @@ namespace MueLu {
       // prolongation factory is in prolongation mode
       Set(coarseLevel, "P", P_smoothed);
 
+      // RfromPFactory used to indicate to TogglePFactory that a factory
+      // capable  or producing R can be invoked later. TogglePFactory
+      // replaces dummy value with an index into it's array of prolongators
+      // pointing to the correct prolongator factory. This is later used by
+      // RfromP_Or_TransP to invoke the prolongatorfactory in RestrictionMode
+      int dummy = 7;
+      Set(coarseLevel,"RfromPfactory", dummy);
```

Comment:
remind me; what does `dummy` do?

Reply:
> remind me; what does `dummy` do?
So this is a little bad but the value of dummy is not used. The toggle factor will just check if "RfromPfactory" is set by someone and then toggle factory will then reset the value associated with RfromPfactory to a value indicating which prolongator within the list of toggle prolongators corresponds to PgP Factory.  There should be comments within the toggle factory.